src/read_write/ply.rs

```diff
@@ -408,48 +407,117 @@ impl PlyIterator {
             reader: BufReader::with_capacity(num_bytes_per_point * 1024, file),
             readers,
             num_total_points: header["vertex"].count,
+            batch_size,
             offset: header.offset,
             point_count: 0,
         })
     }
 }
 
+fn to_vec<T>(slice: &[u8], cur_batch_size: usize) -> Vec<T>
+where
+    T: Clone,
+{
+    let ptr = slice.as_ptr() as *const T;
+    unsafe { std::slice::from_raw_parts(ptr, cur_batch_size).to_vec() }
+}
+
+fn batch_from_readers(
+    readers: &[PropertyReader],
+    batch_size: usize,
+    offset: &Vector3<f64>,
+) -> PointsBatch {
+    let (mut x_vec, mut y_vec, mut z_vec) = (Vec::new(), Vec::new(), Vec::new());
+    let (mut r_vec, mut g_vec, mut b_vec) = (Vec::new(), Vec::new(), Vec::new());
+    let mut attributes = BTreeMap::new();
+    for reader in readers {
+        match &reader.prop.name as &str {
+            "x" => x_vec = to_vec::<f64>(&reader.buf, batch_size),
+            "y" => y_vec = to_vec::<f64>(&reader.buf, batch_size),
+            "z" => z_vec = to_vec::<f64>(&reader.buf, batch_size),
+            "r" | "red" => r_vec = reader.buf[..batch_size].to_vec(),
```

Comment:
Can we read other "vector" attributes than color and position? If so, where is that defined?

Reply:
Ah I see, you answered that above.

Reply:
Is that a TODO? If so, consider adding a TODO comment.

Reply:
Done.